### PR TITLE
fix(ci): pin versions of sphinx dependencies to unlock ci

### PR DESCRIPTION
**Pull request by christophe-papazian**
Many sphinx dependencies were just updated with patch versions that drop sphinx 4.x compatiblity. 
Pins those versions to unlock the ci.

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)
- [x] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

### CODE REVIEW

**Review by gnufede — APPROVED** [review]

File: docs/conf.py [hunk 1]
```diff
@@ -124,7 +124,7 @@ def _skip(self, word):
 #
 # This is also used if you do content translation via gettext catalogs.
 # Usually you set "language" from the command line for these cases.
-language = None
+language = "en"
```
Comment (romainkomorndatadog) [inline docs/conf.py:127]: Is this still necessary, even with the dependencies pinning?

**Review by romainkomorndatadog — APPROVED** [review]
Tentatively approving this, but I think we should at the very least have a follow-up effort to get off of a almost-2-year-old version pin.

File: docs/conf.py (hunk 1 above)
Comment (gnufede) [reply docs/conf.py:127]: Indeed, not needed

Comment (gnufede) [inline docs/conf.py:127]:
```suggestion
language = None
```

**Pull request merged**